public/components/health-check/services/check-pattern.service.ts

```diff
@@ -11,92 +11,24 @@
-        await SavedObject.createWazuhIndexPattern(defaultPatternId);
-        checkLogger.action(`Created index pattern [${defaultPatternId}]`);
-      }
-      checkLogger.info(`Getting list of valid index patterns [${patternId}]...`);
-      listValidIndexPatterns = await SavedObject.getListOfWazuhValidIndexPatterns(defaultIndexPatterns, HEALTH_CHECK);
-      checkLogger.info(`Valid index patterns found: ${listValidIndexPatterns.length || 0}`);
-      if(!AppState.getCurrentPattern()){
-        AppState.setCurrentPattern(defaultPatternId);
-        checkLogger.info(`Index pattern set in cookie: [${defaultPatternId}]`);
-      }
-    } catch (error) {
-      AppState.removeCurrentPattern();
-      checkLogger.info(`Removed current pattern from cookie: [${patternId}]`);
-
-      throw error;
-    }
-  }
-
-  if (AppState.getCurrentPattern() && listValidIndexPatterns.length) {
-    const indexPatternToSelect = listValidIndexPatterns.find(item => item.id === AppState.getCurrentPattern());
-    if (!indexPatternToSelect){
-      AppState.setCurrentPattern(indexPatternToSelect.id);
-      checkLogger.action(`Set index pattern id in cookie: [${indexPatternToSelect.id}]`);
-    };
-  }
-  
-  checkLogger.info(`Checking the app default pattern exists: id [${defaultPatternId}]...`); 
-  const existsDefaultPattern = await SavedObject.existsIndexPattern(defaultPatternId);
-  checkLogger.info(`Default pattern with id [${defaultPatternId}] exists: ${existsDefaultPattern.status ? 'yes' : 'no'}`); 
-  
-  existsDefaultPattern.status
-    && getDataPlugin().indexPatterns.setDefault(defaultPatternId, true)
-    && checkLogger.action(`Default pattern id [${defaultPatternId}] set as default index pattern`);
-
-  patternId && checkLogger.info(`Checking the index pattern id [${patternId}] exists...`);
-  const patternData = patternId ? (await SavedObject.existsIndexPattern(patternId)) || {} : {} ;
-  patternId && checkLogger.info(`Index pattern id exists [${patternId}]: ${patternData.status ? 'yes': 'no'}`);
-
-  if (!patternData.status) {
-    if (listValidIndexPatterns.length) {
-      const indexPatternDefaultFound = listValidIndexPatterns.find((indexPattern) => indexPattern.title === defaultPatternId);
-      checkLogger.info(`Index pattern id exists [${defaultPatternId}]: ${indexPatternDefaultFound ? 'yes': 'no'}`);
-      if(indexPatternDefaultFound){
-        AppState.setCurrentPattern(indexPatternDefaultFound.id);
-        checkLogger.action(`Index pattern set in cookie: [${indexPatternDefaultFound.id}]`);
-      }
-      checkLogger.info('Retring the check...');
-      return await checkEventsPattern(appConfig, checkLogger);
-    } else {
-      checkLogger.error('The selected index-pattern is not present');
-    }
+import {
+  checkAlerts,
+  checkFieldsService,
+  checkTemplateService
+} from './index';
+
+export const checkPatternService = (appConfig) => async (checkLogger: CheckLogger) =>  await checkPattern(appConfig, checkLogger);
```

Comment:
I think `checkIndexPattern` is a better name for this service and `check-index-pattern.service ` for the file

Reply:
Changed